src/battle_util.c

```diff
@@ -11250,11 +11283,18 @@ bool8 AreBattlersOfOppositeGender(u8 battler1, u8 battler2)
     return (gender1 != MON_GENDERLESS && gender2 != MON_GENDERLESS && gender1 != gender2);
 }
 
-u32 CalcSecondaryEffectChance(u8 battler, u8 secondaryEffectChance)
+u32 CalcSecondaryEffectChance(u8 battler, u16 move)
 {
-    if (GetBattlerAbility(battler) == ABILITY_SERENE_GRACE)
+    u8 secondaryEffectChance = gBattleMoves[move].secondaryEffectChance;
+    bool8 hasSereneGrace = (GetBattlerAbility(battler) == ABILITY_SERENE_GRACE);
```

Comment:
If you're going to make a variable for `hasSereneGrace`, would be cleaner if you made a variable `hasRainbow = (gSideStatuses[GetBattlerSide(battler)] & SIDE_STATUS_RAINBOW) != 0` also.

Reply:
Yeah, this is much better. Will do.